### DISCUSSION TITLE
Information on temporal departures

### DISCUSSION THREAD

**Discussion by gkb999** [xCDAT/xcdat#538] [2. Questions] [answered]
Greetings
From the page : https://xcdat.readthedocs.io/en/stable/generated/xarray.Dataset.temporal.departures.html
Departures calculated say: if no Reference_period is provided, the climatological reference period will be the full period covered by the dataset. In that case, the anomaly should be zero, right?
I used that option, where I left xcdat to use the default way to calculate anomalies and I don't get zero yet.
Any explanation would be appreciated

**Comment by pochedls** [comment, marked as answer]
@gkb999 - I don't think the anomaly should necessarily be zero. If I have 6 temperature values: `[255, 260, 265, 265, 260, 255]` and I use the full period (all values) then the mean is `260`. I subtract the mean from each value to get the departures: `[-5, 0, 5, 5, 0, -5]`. So there are zero values, but the anomalies are not always zero.

**Reply by gkb999** [reply]
Thanks a lot!! It answers what I want

**Reply by gkb999** [reply]
Also, does adding weighted to "True" strengthens the non-zero justification?

**Comment by tomvothecoder** [comment]
I converted this from a GH Issue to a Discussion Q&A post.

**Reply by gkb999** [reply]
@pochedls 
Sorry to bring this back after a while.
Yet, again, if I change the baseline/reference period, I get different results.
Like my data was from 2002-2022
I used 2002-2012, got a different one, 2005-2015, got a different one and just used as below:
The one without any baseline/reference period: like: ```s_depart = ds_climo.temporal.departures("ts", "month")``` and got a different result. 
The last default one, uses some baseline (climatology) to compute right?

Like I'm using 20 years of data, just curious if it uses all the 20 years as climatological average? if yes, then I should get 'zero'.
It might be using a default baseline based on the no.of samples?
Can you please explain?

**Reply by pochedls** [reply]
@gkb999 - please look at the [previous post](https://github.com/xCDAT/xcdat/discussions/538#discussioncomment-6844693) – you should not expect to get anomalies of zero if you use all data. For the following time series of pretend annual values: 

`[255, 260, 265, 265, 260, 255]`

| Base period | Climatological Mean | Annual Anomalies |
| - | - | - |
| All years | `260.` | `[-5.,  0.,  5.,  5.,  0., -5.]` |
| First 3 years | `260.` | `[-6., -1.,  4.,  4., -1., -6.]` |
| First 5 years | `261.` | `[-5.,  0.,  5.,  5.,  0., -5.]` |

Note the departures are different if you use three versus five years as the climatological period.
